### PR TITLE
Instantiate parser from SpreadsheetCalculation

### PR DESCRIPTION
**Pull request by khameeteman**

### CODE REVIEW

File: excel_graph_parser/graph_parser.py [hunk 1]
```diff
@@ -248,8 +260,8 @@ def get_figure_titles(self):
         return figure_list
 
     @staticmethod
-    def create_ploty_figure(chart_data: dict):
-        """Creates ploty figure based on the extracted chart data"""
+    def _create_plotly_figure(chart_data: dict) -> dict:
```
Comment (khameeteman) [inline excel_graph_parser/graph_parser.py:263]: Note that this breaks compatibility if actually used in apps, which I highly doubt

File: excel_graph_parser/graph_parser.py [hunk 2]
```diff
@@ -216,15 +229,14 @@ def validate_sheet_names(self):
         """Validate that the input sheet and output sheets are present"""
         wb = self.workbook
         if not all(sheetname in wb for sheetname in ["viktor-input-sheet", "viktor-output-sheet"]):
-            os.unlink(self.excel_file_path)
```
Comment (kvangiessen-VIKTOR) [inline excel_graph_parser/graph_parser.py:219]: Why did you remove this?

File: excel_graph_parser/graph_parser.py (hunk 1 above)
Comment (kvangiessen-VIKTOR) [reply excel_graph_parser/graph_parser.py:263]: I think we should be fine.

**Review by kvangiessen-VIKTOR — COMMENTED** [review]
Looks pretty good, but Im not yet sure how to test this properly.

File: excel_graph_parser/graph_parser.py (hunk 2 above)
Comment (khameeteman) [reply excel_graph_parser/graph_parser.py:219]: I don't think the parser should be responsible for removing files. If I have a static spreadsheet and pass its path to the parser, the parser will remove my file which is not expected

Comment (kvangiessen-VIKTOR) [reply excel_graph_parser/graph_parser.py:219]: Makes sense!

**Review by kvangiessen-VIKTOR — APPROVED** [review]
